### PR TITLE
[enh] Switch to alt (from shift) for slicing (render) plane drag binding

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# References and relevant issues
Came up while working with an end-user
https://napari.zulipchat.com/#narrow/stream/212875-general/topic/clipping.20planes.20and.20slicing.20planes/near/439290086

# Description
Prior to this PR, Shift-drag was used to drag the slicing (render) plane along its normal.
But, the vispy camera translation also uses Shift-drag. The clash can make it hard to position a slicing (render) plane, particularly when there is a volume rendering shown at the same time.

As a result, in this PR I change the keybinding to use Alt instead of Shift. This permits dragging the plane or translating the camera/viewer.
I also update the slicing (render) planes example.
This is technically a breaking change, but slicing (render) planes were not documented.

### CODE REVIEW

**Review by brisvag — APPROVED** [review]
Just checking: `Alt` vs `Meta`, are we good on that side?

Generally, love this change, I always have the same issue as well, quite frustrating.

**Comment by psobolewskiPhD** [comment]
Yeah, sorry mac centric approach where alt is used for these "same thing but do something different" actions.
I can change to meta if that's better on the other platforms.

**Comment by brisvag** [comment]
I have no clue, I asked because I thought you encountered this more xD @Czaki maybe knows if this can be an issue?

**Comment by Czaki** [comment]
looks ok for me

**Comment by psobolewskiPhD** [comment]
The user pointed out to me that she uses napari-threedee point-on-a-plane annotator and that uses alt-click. 
I fear that this change will make a point appear when you first alt-click to drag.
I will test.

**Comment by Czaki** [comment]
Or maybe make it configurable from settings?

**Comment by alisterburt** [comment]
Hi guys - I don't think this PR should be merged. Alt + click and drag is used for window dragging in some Linux flavours commonly used in universities (alma) and can not be easily reconfigured by end users

**Comment by psobolewskiPhD** [comment]
Converting to draft to prevent accidental merge.
I appreciate the feedback @alisterburt
Revisiting my goal with this, is being able to move the volume (really the camera) when zoomed in and have a render plane.
One solution that Alister suggested in zulip would be have space bar work in 3D like in 2D.
This would suffice to disambiguate the actions in most use cases I think.
Alas, as noted above the volume drag is really vispy camera move and it's not clear to me how one would try to bind space bar to the behavior in addition to shift.

**Comment by brisvag** [comment]
Yeah the `alt-click` conflict with window drag is unfortunate, it blocks out a very simple and intuitive shortcut. Are there any concerns about `Ctrl+click`? That's unused as far as I can tell.

**Comment by psobolewskiPhD** [comment]
I'll check ctrl. On macOS it mimics the other click -- right-click -- but this may be a non-issue if we dont plan on canvas contextual menus anytime soon.

**Comment by psobolewskiPhD** [comment]
Yeah changing it to `Ctrl` makes it not work on my mac. Same with `Meta`.
So it seems we're at an impasse.

**Comment by brisvag** [comment]
ctrl+shit? :sweat_smile:

**Comment by psobolewskiPhD** [comment]
> ctrl+shit? 😅

Um... 😶 
Not that kind of impasse?
🤣 
I'll try Ctrl-Shift and Alt-Shift
On mac Alt-Shift should work, but would that also trigger the linux issue?

**Comment by psobolewskiPhD** [comment]
Oh, derp, so `Control` in event.modifiers on macOS is actually Meta (Command) on keyboard.
So `Control` alone on macOS does work, you just need to use Command drag!
I get why we switch it for most *keybindings*, but I wonder if there is a way to make it just Control on all platforms here, as a mouse modifier.

**Comment by brisvag** [comment]
Ooof... thanks for the deep dive @psobolewskiPhD :sweat_smile: I would go for ctrl/command then, though I don't know what's the etiquette for command shortcuts on mac. If that's fine, it's definitely better than meta, and at least is just an "upgrade" of what we had before so it's a bit intutive for those who were using it already.

**Comment by alisterburt** [comment]
Is there really a need to change this? Could you please leave the existing keybinding in place so existing workflows aren't interrupted?

I think the 'terribleness' of this shortcut is overstated and the correct napari solution is to make 2D + 3D behaviour both use space bar rather than changing this

**Comment by alisterburt** [comment]
Also want to add, this is a feature which has been in napari for years and that people have built packages around - please don't do this in a patch release if you do it

**Comment by psobolewskiPhD** [comment]
Bleh, sorry. I guess I just have a lot more "memory" for the vispy/volume/camera shift-drag than the render plane one.

I'll try to think of some other solution for this -- maybe we can make the modifier a setting? Not sure if the app-model key bind system supports a mouse modifier.

BTW We'll have the same issue with clipping planes -- I'd like to make them draggable as well.

**Comment by brisvag** [comment]
It's true that it's been around for a long time, but it's also true that the problem it causes has been bad since day 1 ^^', and when this shorctut was added we did it without much consideration and simply borrowed it from @alisterburt 's example (https://github.com/napari/napari/blob/main/examples/3Dimage_plane_rendering.py) which initially used shift.

Having said that, agree 100% that this is not patch release territory, and that we must clearly communicate this in the highlights because a lot of people use this.

I don't quite follow the spacebar argument: spacebar is the shorcut to switch temporarily layer mode to `pan_zoom`. This won't do anything, unless we either:
- change the `plane` rendering into a "mode" instead of a rendering method
- overload the `space` shortcut so when in 3D it not only cahnges mode but also disables plane moving? But what if someone whats to go to panzoom specifically to move the plane??

**Comment by alisterburt** [comment]
I'm suggesting changing so that space bar activates panning in 3D which is more consistent than having it be space bar in 2D and shift in 3D (which is only the case because it's default in vispy)

This would also leave more modifier keys available to users for custom stuff

**Comment by brisvag** [comment]
I guess you're referring to the workflow where one select a labels layer (not images) and uses spacebar to switch to pan_zoom mode from paint mode temporarily.

But in practice, spacebar does not "activate panning" in 2D. Panning in 2D is just `left-click`, with no modifiers. In 3D, panning is `shift+left-click` because `left-click` alone is reserved for camera rotation (which is consistent most other 3D vis software).

What Spacebar does is change layer mode to pan_zoom (if the mode is something else, like in the labels painting case).

So basically you're suggesting to eother overload spacebar to mean "switch to pan_zoom and also if it's an Image disable the shortcut for moving the plane", or to change the default (and imo sensible) vispy 3D controls so that panning is just `click+drag`, nothing is bound to `shift+drag` and rotation is something else?

**Comment by psobolewskiPhD** [comment]
To followup on the discussion here:
- clipping planes can coexist with rendering planes: what modifier should be used to drag them?

**Comment by TimMonko** [comment]
I see that Lorenzo just removed the milestone, but now this proposal will conflict with this issue as well #8276

**Comment by brisvag** [comment]
At some point we need to do a proper sit-down and figure out all the shortcuts defaults we want and lock in :P
